### PR TITLE
add BFloat16 support for BatchNorm on CPU

### PR DESCRIPTION
**Pull request by mingfeima**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77496

Differential Revision: [D36404028](https://our.internmc.facebook.com/intern/diff/D36404028)

### CODE REVIEW

**Comment by mingfeima** [comment]
Reland https://github.com/pytorch/pytorch/pull/74410, failure has been removed! @frank-wei, @albanD, please review!

The old failure from https://github.com/pytorch/pytorch/runs/6433364399?check_suite_focus=true is due to:
1. `test_nnc_correctness` on batch_norm has [a bug with empty input](https://github.com/pytorch/pytorch/issues/71286), and the test case is marked as expected failure now at [L13705@common_methods_invocations.py](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L13705)
2. nnc fuser is not enabled with `bfloat16`, [link](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L937), so the test case `test_nnc_correctness_nn_functional_batch_norm_cpu_bfloat16` is not comparing the fused kernel at all and therefore it passed. At the moment, mark the test case as 'skipped'.

All the slow CI passed now.

**Comment by frank-wei** [comment]
> Reland #74410, failure has been removed! @frank-wei, @albanD, please review!
> 
> The old failure from https://github.com/pytorch/pytorch/runs/6433364399?check_suite_focus=true is due to:
> 
> 1. `test_nnc_correctness` on batch_norm has [a bug with empty input](https://github.com/pytorch/pytorch/issues/71286), and the test case is marked as expected failure now at [L13705@common_methods_invocations.py](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L13705)
> 2. nnc fuser is not enabled with `bfloat16`, [link](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L937), so the test case `test_nnc_correctness_nn_functional_batch_norm_cpu_bfloat16` is not comparing the fused kernel at all and therefore it passed. At the moment, mark the test case as 'skipped'.
> 
> All the slow CI passed now.

Thanks for the investigation. The NNC is not ready with bfloat16 yet. I think it is ok to mark it as skipped. Ask @albanD for 2nd opinion.

**Comment by frank-wei** [comment]
@frank-wei has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36404028).

**Review by frank-wei — APPROVED** [review]
